plugins/inputs/redfish/redfish.go

```diff
@@ -0,0 +1,358 @@
+	UpperThresholdCritical float64
+	UpperThresholdFatal    float64
+	Status                 VoltStatus
+}
+type VoltStatus struct {
+	State  string
+	Health string
+}
+type Address struct {
+	PostalAddress PostalAddress
+	Placement     Placement
+}
+type PostalAddress struct {
+	DataCenter string
+	Room       string
+}
+type Placement struct {
+	Rack string
+	Row  string
+}
+type Redfish struct {
+	Host              string `toml:"host"`
+	BasicAuthUsername string `toml:"username"`
+	BasicAuthPassword string `toml:"password"`
+	Id                string `toml:"id"`
+	client            http.Client
+	tls.ClientConfig
+	Timeout internal.Duration `toml:"timeout"`
+}
+
+func (r *Redfish) Description() string {
+	return "Read CPU, Fans, Powersupply and Voltage metrics of Dell/HP hardware server through redfish APIs"
+}
+
+var redfishConfig = `
+  ##Server  OOB-IP
+  host = "192.0.0.1"
+
+  ##Username,  Password   for   hardware   server
+  username = "test"
+  password = "test"
+
+  ##Resource  Id   for   redfish   APIs
+  id="System.Embedded.1"
+
+  ##Optional TLS   Config, if not provided insecure skip verifies defaults to true
+  #tls_ca = "/etc/telegraf/ca.pem"
+  #tls_cert = "/etc/telegraf/cert.pem"
+  #tls_key = "/etc/telegraf/key.pem"
+  
+
+  ## Amount   of   time   allowed   to   complete   the   HTTP   request
+  # timeout = "5s"
+`
+
+func (r *Redfish) SampleConfig() string {
+	return redfishConfig
+}
+
+func (r *Redfish) Init() error {
+	tlsCfg, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	if (len(r.ClientConfig.TLSCA) == 0) || (len(r.ClientConfig.TLSCert) == 0 && len(r.ClientConfig.TLSKey) == 0) {
+		var insecuretls tls.ClientConfig
+		insecuretls.InsecureSkipVerify = true
+		tlsCfg, err = insecuretls.TLSConfig()
+		if err != nil {
+			return err
+		}
+	}
+	r.client = http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: r.Timeout.Duration,
+	}
+	return nil
+}
+
+func Severity(critical, fatal, val int) string {
+	severity := "NA"
+	if (critical != 0) || (fatal != 0) {
+		if (val >= fatal) && (fatal != 0) {
+			severity = "Fatal"
+		} else if (val >= critical) && (critical != 0) {
+			severity = "Critical"
+		} else {
+			severity = "OK"
+		}
+	}
+	return severity
+}
+
+func (r *Redfish) Gather(acc telegraf.Accumulator) error {
+	var url []string
+	var payload Payload
+	url = append(url, fmt.Sprint("https://", r.Host, "/redfish/v1/Chassis/", r.Id, "/Thermal"), fmt.Sprint("https://", r.Host, "/redfish/v1/Chassis/", r.Id, "/Power"), fmt.Sprint("https://", r.Host, "/redfish/v1/Systems/", r.Id), fmt.Sprint("https://", r.Host, "/redfish/v1/Chassis/", r.Id, "/"))
+
+	if len(r.Host) == 0 || len(r.BasicAuthUsername) == 0 || len(r.BasicAuthPassword) == 0 {
+		return fmt.Errorf("Did not provide IP or username and password")
+	}
+	if len(r.Id) == 0 {
+		return fmt.Errorf("Did not provide all the ID of the resource")
+	}
```

Comment:
updated to check Host, username/password and ID in the Init() function(included username/password check as well because for redfish APIs the authorisation is mandatory)